h2o-k8s/src/main/java/water/k8s/KubernetesEmbeddedConfig.java

```diff
@@ -35,6 +37,9 @@ public void notifyAboutEmbeddedWebServerIpPort(InetAddress ip, int port) {
     @Override
     public void notifyAboutCloudSize(InetAddress ip, int port, InetAddress leaderIp, int leaderPort, int size) {
         Log.info(String.format("Created cluster of size %d, leader node IP is '%s'", size, leaderIp.toString()));
+        if (size == cloudSize) {
+            cloudingFinished();
```

Comment:
This change has complex consequences. A normal deployment via "headless service" clustering (not assisted clustering with operator) looks like this: https://github.com/h2oai/h2o-3/tree/master/h2o-k8s - this is the YAML recipe given to our users. There is no persistent volume bound by default when the clustering is done via h2o-k8s module. Which means, **once the pod is down, all data inside the pod are lost**. Helm chart uses the same recipe by default at the moment. 

We have multiple choices:
1. Add PersistentVolume to the documentation + the Helm chart and tell ours users to use this version of YAML. And make this enabled by default. This means, if the user accidentally doesn't define a PersistentVolume for the recovery folder and autorecovery is enabled by default, it won't work, but the client will signal it does.
2. Also add PersistentVolume + the Helm chart and disable this by default. The new YAML will use a dedicated ENV VAR to enable this by default. This will prevent false expectations and failures when newer version of H2O jar is used in combination with older YAML recipe/Helm chart.
3. Leave this disabled by default and support this only via the assisted clustering. Not my preferred option.

I think the ideal solution is as follows:
Add env var to enable/disable autorecovery. Would be present in both `h2o-k8s` and `h2o-clustering` module. In both modules, it will have to be explicitly disabled. In assisted clustering, it makes sense, as there is "expert software" that knows how to operate H2O and has no problem with distinguishing different use cases. In case of h2o-k8s module, this is to counter false expectations when combined with older YAML/Helm.

Reply:
auto-recovery is enabled only when -auto_recovery_dir is set, so I dont think this is a problem